core/src/main/java/io/questdb/griffin/engine/functions/StrFunction.java

```diff
@@ -63,6 +63,9 @@ public final byte getByte(Record rec) {
 
     @Override
     public final char getChar(Record rec) {
+        CharSequence val = getStr(rec);
+        if (val != null && val.length() == 1) return val.charAt(0);
+
         throw new UnsupportedOperationException();
```

Comment:
if this is to support `cast` then we should not be throwing unsupported operation exception. This might cause a nasty unhandled error rather than perhaps failing more gracefully?

Reply:
Ok, I reverted this change, not essential